internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -0,0 +1,56 @@
+// Package codepipeline provides a client to make API requests to Amazon Elastic Container Service.
+package codepipeline
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	cp "github.com/aws/aws-sdk-go/service/codepipeline"
+)
+
+type codepipelineClient interface {
+	GetPipeline(*cp.GetPipelineInput) (*cp.GetPipelineOutput, error)
+}
+
+// CodePipeline wraps the AWS CodePipeline client.
+type CodePipeline struct {
+	client codepipelineClient
+}
+
+// Pipeline contains information about the pipeline
+// TODO wrap nested resources or just use what the SDK provides?
```

Comment:
We don't want to pass directly what the SDK provides otherwise a client that imports our `codepipeline` package will also need to import the SDK's `codepipeline` pkg causing a name collision which we want to avoid.

So we can remove this comment.

Reply:
Sorry, what i meant here was do we want to create every type as a direct mapping such as `type ArtifactStore cp.ArtifactStore` or do we want to pare down the API the sdk resource. like what I curently do with the `Pipeline` custom type. 

I was going to save most of the work fleshing out the types here for when I imlement `execute` so it was just more of a note for myself but i can remove it.